### PR TITLE
DEMO++ field cage

### PR DESCRIPTION
**Pull request by paolafer**
This PR reviews the geometry of the DEMO++ field cage and updates it to the latest drawings.

### CODE REVIEW

File: source/geometries/NextDemoFieldCage.h [hunk 1]
```diff
@@ -94,104 +70,46 @@ namespace nexus {
...
-    const G4double anode_ring_thickn_;
-    const G4double anode_ring_length_;
-    const G4double cathode_ring_height_;
-    const G4double cathode_ring_diam_;
-    const G4double cathode_ring_thickn_;
-    const G4double tpb_thickn_;
-
-    const G4double ltube_diam_;
+    const G4double light_tube_drift_start_z_, light_tube_drift_end_z_;
     const G4double ltube_thickn_;
```
Comment (jmunozv) [inline source/geometries/NextDemoFieldCage.h]: You are using two different names to refer to the same volume: "ltube_XXXX" and "light_tube_XXXX".
Please select just one.

File: source/geometries/NextDemoFieldCage.cc [hunk 2]
```diff
@@ -39,890 +33,567 @@
...
+      new G4LogicalVolume(gate_grid_solid, grid_mat, "GATE_GRID");
+
+    G4double grid_zpos = el_gap_length/2. - grid_thickn_/2.;
+    new G4PVPlacement(0, G4ThreeVector(0., 0., grid_zpos),
+                      gate_grid_logic, "GATE_GRID",
+                      elgap_logic, false, 0, false);
+
+    G4LogicalVolume* anode_logic = nullptr;
+    if (plate_) {
+      G4Tubs* anode_quartz_solid =
```
Comment (jmunozv) [inline source/geometries/NextDemoFieldCage.cc:385]: You have implemented the quartz volume, but not  ITO and TPB coatings.

File: source/geometries/NextDemoFieldCage.cc [hunk 3]
```diff
@@ -39,890 +33,567 @@
...
+    num_rings_ (23),
+    ring_diam_ (235. * mm),
+    ring_height_ (10. * mm),
+    ring_thickn_ (5. * mm),
+    ring_drift_buff_gap_ (27.2 * mm),
+    dist_drift_ring_centres_ (15. * mm),
+    dist_buff_ring_centres_ (20. * mm),
+    dist_gate_first_ring_ (18.4 * mm),
+    bar_start_z_ (14. * mm),
+    bar_end_z_ (403.5 * mm),
```
Comment (jmunozv) [inline source/geometries/NextDemoFieldCage.cc]: I have a value of 413.5 mm for this dimension in my notes. Could you please check it ??

**Review by jmunozv — CHANGES_REQUESTED** [review]

File: source/geometries/NextDemoFieldCage.h (hunk 1 above)
Comment (paolafer) [reply source/geometries/NextDemoFieldCage.h]: Done.

File: source/geometries/NextDemoFieldCage.cc (hunk 2 above)
Comment (paolafer) [reply source/geometries/NextDemoFieldCage.cc:385]: Done.

**Comment by jmunozv** [comment]
I approve this first version of NextDEMO Field Cage

**Review by jmunozv — APPROVED** [review]

**Pull request merged**